include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -553,21 +607,7 @@ namespace VectorTools
     VectorType &                                               vec,
     const ComponentMask &                                      component_mask)
   {
-    Assert(dof_handler.get_fe().n_components() == function.n_components,
-           ExcDimensionMismatch(dof_handler.get_fe().n_components(),
```

Comment:
I don't quite understand why you need to duplicate this function into one for the hp and one for the non-hp case. Can't you just replace this call to `get_fe()` by `get_fe_collection()`? That works for both `::DoFHandler` and `hp::DoFHandler`. If you did that, then you wouldn't have to duplicate the body of the function...

Reply:
Done!